src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -593,6 +626,33 @@ private void WriteEventToDatabase(LogEventInfo logEvent)
             }
         }
         /// <summary>
+        /// Set Parameter Type and Value
+        /// </summary>
+        protected void SetParameterInfo(IDbDataParameter p, DatabaseParameterInfo par, string value)
+        {
+            EnsureResolveParameterInfo(p);
+            this.ParameterConverter.SetParameterDbType(p, par);
+            this.ParameterConverter.SetParameterValue(p, par, value);
+        }
+        /// <summary>
+        /// Resolve Parameter DbType And Value Converter
+        /// </summary>
+        protected void EnsureResolveParameterInfo(IDbDataParameter p)
+        {
+            if (this.ParameterConverter == null)
+            {
+                lock (this.SyncRoot)
+                {
+                    if (this.ParameterConverter == null)
+                    {
+                        var converter = (DatabaseParameterConverter)Activator.CreateInstance(this.ParameterConverterType);
```

Comment:
as proposed: replace with virtual call

Reply:
class 'DatabaseTarget' don't need to change, we only inherit from 'DatabaseParameterConverter' to support more DbType.